src/compiler/checker.ts

```diff
@@ -26642,7 +26650,14 @@ namespace ts {
         }
 
         function getSuggestedSymbolForNonexistentProperty(name: Identifier | PrivateIdentifier | string, containingType: Type): Symbol | undefined {
-            return getSpellingSuggestionForName(isString(name) ? name : idText(name), getPropertiesOfType(containingType), SymbolFlags.Value);
+            let props = getPropertiesOfType(containingType);
+            if(typeof name !== "string") {
+                const parent = name.parent;

```

Comment:
You can reuse the `typeof` check above to avoid the `isString(...)` check below.

```suggestion
                const parent = name.parent;
                name = idText(name);
```